cadc-util/src/main/java/ca/nrc/cadc/xml/XmlUtil.java

```diff
@@ -260,32 +253,50 @@ public static SAXBuilder createBuilder(Map<String, String> schemaMap)
         }
 
         SAXBuilder builder = new SAXBuilder(rf, sh, jf);
-        if (schemaVal)
-        {
+        if (schemaVal) {
             builder.setFeature("http://xml.org/sax/features/validation", true);
             builder.setFeature("http://apache.org/xml/features/validation/schema", true);
-            if (schemaMap.size() > 0)
+            if (schemaMap.size() > 0) {
                 builder.setProperty("http://apache.org/xml/properties/schema/external-schemaLocation",
-                    sbSchemaLocations.toString());
+                                    sbSchemaLocations.toString());
+            }
         }
         return builder;
     }
 
     /**
      * Get an URL to a schema file. This implementation finds the schema file using the ClassLoader
      * that loaded the argument class.
-     * 
+     *
      * @param resourceFileName
-     * @param runningClass 
+     * @param runningClass
      * @return
      */
-    public static String getResourceUrlString(String resourceFileName, Class<?> runningClass)
-    {
+    public static String getResourceUrlString(String resourceFileName, Class<?> runningClass) {
         String rtn = null;
         URL url = runningClass.getClassLoader().getResource(resourceFileName);
-        if (url == null)
-            throw new MissingResourceException("Resource not found: " + resourceFileName, runningClass.getName(), resourceFileName);
+        if (url == null) {
+            throw new MissingResourceException("Resource not found: " + resourceFileName, runningClass.getName(),
+                                               resourceFileName);
+        }
         rtn = url.toString();
         return rtn;
     }
+
+    /**
+     * Obtain whether the given Document contains a Comment containing the given string.
+     *
+     * @param element      The element to check.
+     * @param commentValue The value to verify exists.
+     * @return True if one or more comments match, or False otherwise.
+     */
+    public static boolean hasCommentContaining(final Element element, final String commentValue) {
+        return !element.getContent(new ContentFilter(ContentFilter.COMMENT) {
+            @Override
+            public Content filter(final Object obj) {
+                final Content c = super.filter(obj);
+                return ((c != null) && c.getValue().contains(commentValue)) ? c : null;
+            }
+        }).isEmpty();
+    }
```

Comment:
I find this difficult to read with the ! at the start of the expression and the isEmpty() at the end.  I think it would be more maintainable if it were split into separate lines of logic.

But in general, I wonder if this type of utility is useful in a core library?  The method is quite specific and and I can't really see it being used outside of the integration test for client IP support.  So, I'd suggest moving it to the actual test.

Reply:
It is used in both `cadc-test-vosi` and `cadc-vosi`.  The `XmlUtil` just seemed like a natural place to put it.  Is there a shared Test class?

Reply:
I don't see it being used in cadc-vosi (and it shouldn't be) because cadc-test-vosi should be the only thing that needs to do this check in the integration tests.  vosi itself doesn't have to read comments from XML.

Reply:
There is a `src/test/java/ca/nrc/cadc/vosi/AvailabilityTest` (unit test) in `cadc-vosi`, and a `src/test/java/ca/nrc/cadc/vosi/AvailabilityTest` (integration test) in `cadc-test-vosi`.  Do we only want to support one of those?